tests/eas/capacity_capping.py

```diff
@@ -109,6 +131,33 @@ def populate_params(cls):
             ).get()
 
     @classmethod
+    def _freeze_userspace(cls):
+        if "freeze_userspace" not in TEST_CONF.get("flags", []):
+            cls._log.info('not asked to freeze userspace');
+            return
+        if 'cgroups' not in cls.env.target.modules:
+            raise RuntimeError(
+                'Failed to freeze userspace. Ensure "cgroups" module is listed '
+                'among modules in target/test configuration')
+        controllers = [s.name for s in cls.env.target.cgroups.list_subsystems()]
+        if 'freezer' not in controllers:
+            cls._log.warning('No freezer cgroup controller on target. '
+                              'Not freezing userspace')
+            return
+
+        exclude = cls.critical_tasks[cls.env.target.os]
+        cls._log.info('Freezing all tasks except: %s', ','.join(exclude))
+        cls.env.target.cgroups.freeze(exclude)
+        cls.userspace_frozen = True
```

Comment:
Agree, was thinking the same above... and regarding this context manager: nice idea. It can perhaps also be useful for stuff like auto-connecting a device across reboots, etc...